packages/core/helper-plugin/src/utils/auth.ts

```diff
@@ -0,0 +1,196 @@
+// TODO @soupette we need to refactor this file
+
+import isEmpty from 'lodash/isEmpty';
+import isNil from 'lodash/isNil';
+
+const TOKEN_KEY = 'jwtToken';
+const USER_INFO = 'userInfo';
+const CURRENT_STEP = 'GUIDED_TOUR_CURRENT_STEP';
+const COMPLETED_STEPS = 'GUIDED_TOUR_COMPLETED_STEPS';
+const SKIPPED = 'GUIDED_TOUR_SKIPPED';
+const THEME_KEY = 'STRAPI_THEME'; // Also used in packages/core/admin/admin/src/components/ThemeToggleProvider/index.js
+const UPLOAD_MODAL_VIEW = 'STRAPI_UPLOAD_MODAL_VIEW';
+const UPLOAD_VIEW = 'STRAPI_UPLOAD_LIBRARY_VIEW';
+
+interface UserInfo {
+  email: string;
+  firstname: string;
+  lastname?: string;
+  username?: string;
+  preferedLanguage?: string;
+  id: number;
+  isActive?: boolean;
+  blocked: boolean;
+  createdAt: Date;
+  updatedAt: Date;
+}
+
+interface StorageItems {
+  userInfo: UserInfo;
+  jwtToken: string;
+  STRAPI_THEME: 'light' | 'dark';
+  GUIDED_TOUR_CURRENT_STEP: string | null;
+  GUIDED_TOUR_COMPLETED_STEPS: string[] | null;
+  GUIDED_TOUR_SKIPPED: boolean | null;
+  STRAPI_UPDATE_NOTIF: boolean | null;
+  STRAPI_UPLOAD_MODAL_VIEW: 0 | 1 | null; // grid or list view
+  STRAPI_UPLOAD_LIBRARY_VIEW: 0 | 1 | null; // grid or list view
+  // FIXME: No idea what this is or what its value is
+  videos: string | null;
+  // FIXME: No idea what this is or what its value is
+  onboarding: string | null;
```

Comment:
Not sure what these are or what their parsed shape looks like...

Reply:
I would do `unknown` perhaps? that we have to cast them and we'll be able to address this issue when we get too it?